////////////////////////////////////////////
//  Core Modules

////////////////////////////////////////////
//  Third Party Modules

////////////////////////////////////////////
//  Third Party Module Instances

////////////////////////////////////////////
//  Third Party Middleware

////////////////////////////////////////////
//  Third Party Config Files

////////////////////////////////////////////
//  My Middleware

////////////////////////////////////////////
//  Routing Middleward

////////////////////////////////////////////
//  My Modules

////////////////////////////////////////////
//  Exported Anonomous Ansync Error Catcher

module.exports = (fn) => (request, response, next) =>
  fn(request, response, next).catch((error) => {
    console.log(error, error.message);
    return next(error);
  });
